### PR TITLE
Add getErrorsIndexedByAttribute() and getCommonErrors() methods to Result class

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | Related - https://github.com/yiisoft/form/issues/177

### CODE REVIEW

File: src/Result.php [hunk 1]
```diff
@@ -72,6 +72,22 @@ public function getAttributeErrorObjects(string $attribute): array
...
 
+    /**
+     * @psalm-return array<int|string, non-empty-list<int|string>>
+     */
+    public function getTopLevelAttributeErrors(): array
+    {
+        $errors = [];
+        foreach ($this->errors as $error) {
+            $firstItem = $error->getValuePath()[0] ?? null;
+            if ($firstItem !== null) {
```
Comment (vjik) [inline src/Result.php:83]: Maybe add `null` errors with key `''` (empty string)?
This is will be consistently with `getErrorsIndexedByPath`.

File: tests/ResultTest.php [hunk 2]
```diff
@@ -102,6 +102,14 @@ public function testGetAttributeErrorObjects(): void
         );
     }
 
+    public function testGetTopLevelAttributeErrors(): void
```
Comment (vjik) [inline tests/ResultTest.php:105]: Additional need test errors without path.

Comment (arogachev) [reply tests/ResultTest.php]: :heavy_check_mark:

File: src/Result.php (hunk 1 above)
Comment (arogachev) [reply src/Result.php]: :heavy_check_mark:

**Review by vjik — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Pull request merged**